mantis-runtime/src/main/java/io/mantisrx/runtime/executor/StageExecutors.java

```diff
@@ -276,41 +276,39 @@ private static <T, R> Observable<Observable<R>> executeInners(Observable<Observa
      * @return untyped to support multiple callers return types
      */
     @SuppressWarnings("unchecked")
-    private static <T, R> Observable<Observable<R>> executeInnersInParallel(Observable<Observable<T>> oo, Computation computation,
-                                                                            final Context context, final boolean applyTimeoutToInners, final long timeout, final int concurrency) {
+    private static <T, R> Observable<Observable<R>> executeInnersInParallel(
+        Observable<Observable<T>> oo,
+        Computation computation,
+        final Context context,
+        final boolean applyTimeoutToInners,
+        final long timeout,
+        final int concurrency) {
         logger.info("initializing {}", computation.getClass().getCanonicalName());
         computation.init(context);
 
         // from groups to observable
         final Func2<Context, Observable<T>, Observable<R>> c
                 = (Func2<Context, Observable<T>, Observable<R>>) computation;
 
-        if (concurrency == StageConfig.DEFAULT_STAGE_CONCURRENCY) {
```

Comment:
In this original code, if the concurrency is 1, then you don't need to do a groupBy and flatMap. Which may be more efficient. I feel we should keep that.